test/Test_ZProc.ml

```diff
@@ -33,8 +34,30 @@ let simplification ~(zpath : string) () =
     let res = simplify z3 "(and (or (>= x 1) (> x 5)) (<= x 9))"
     in Alcotest.(check string) "same" res "(and (<= x 9) (>= x 1))")
 
+let z3_result_to_model_as_array () =
+  let result = ["sat"; "true";
+  "(model 
+                                             (define-fun x () (Array Int Int)
+                                                (_ as-array k!10!12))
+                                              (define-fun y () Int
+                                                10)
+                                                (define-fun k!10!12 ((x!0 Int)) Int
+                                                (ite (= x!0 2) 2
+                                                (ite (= x!0 1) 1
+                                                  3)))
```

Comment:
Can you reduce the indentation for this block? It should only be indented one more level than `(model`